python/GangaLHCb/Lib/Applications/GaudiExec.py

```diff
@@ -196,9 +196,13 @@ def unprepare(self, force=False):
         if self.is_prepared is not None:
             self.decrementShareCounter(self.is_prepared.name)
             self.is_prepared = None
+        if self.getShareCounterVal(self.is_prepared.name)==0 and self.uploadedInput:
```

Comment:
Putting this ability into the SharedDir object would also prevent that the remove operation has to be implemented in multiple places.